src/analysis/forward.jl

```diff
@@ -1,5 +1,65 @@
 using Core.Compiler: StmtInfo, ArgInfo, CallMeta, AbsIntState
 
+if VERSION >= v"1.12.0-DEV.1268"
+
+using Core.Compiler: Future
+
+function fwd_abstract_call_gf_by_type(interp::AbstractInterpreter, @nospecialize(f),
+        arginfo::ArgInfo, si::StmtInfo, sv::AbsIntState, primal_call::Future{CallMeta})
+    if f === ChainRulesCore.frule
+        # TODO: Currently, we don't have any termination analysis for the non-stratified
+        # forward analysis, so bail out here.
+        return Future{Union{CallMeta, Nothing}}(nothing)
+    end
+
+    nargs = length(arginfo.argtypes)-1
+    frule_preargtypes = Any[Const(ChainRulesCore.frule), Tuple{Nothing,Vararg{Any,nargs}}]
+    frule_argtypes = append!(frule_preargtypes, arginfo.argtypes)
+    frule_atype = CC.argtypes_to_type(frule_argtypes)
+
+    local frule_call::Future{CallMeta}
+    local result::Future{Union{CallMeta, Nothing}} = Future{Union{CallMeta, Nothing}}()
+    function make_progress(_, sv)
+        if isa(primal_call[].info, UnionSplitApplyCallInfo)
+            result[] = nothing
```

Comment:
The return type appears to be wrong here. The intended behavior appears to be returning `result[] = primal_call[]` in this case (https://github.com/JuliaDiff/Diffractor.jl/blob/778af00be4cd90d2d14d8c78055e7d060ccd6286/src/stage2/abstractinterpret.jl#L77-L80)

Reply:
That's the function that calls this. Potentially it should be refactored to just do that, but I just wanted to only make the refactoring change.

Reply:
The return type is required to be Future{CallMeta} though, or the caller's caller will be unhappy

Reply:
The call-site there isn't updated yet. This function is called directly from DAECompiler and I adjusted the call-site there to work with `Future{Union{Nothing, CallMeta}}`

Reply:
It is fine if it still branches on r!==nothing there, it will just be dead code now, as it appears you you must handle that case here now, instead of being able to handle it there